### PR TITLE
minor : Deunwrap extract_function

### PR DESCRIPTION
**Pull request by alibektas**
#15398 subtask 5.

### CODE REVIEW

File: crates/ide-assists/src/handlers/extract_function.rs [hunk 1]
```diff
@@ -1934,12 +1941,12 @@ fn fix_param_usages(ctx: &AssistContext<'_>, params: &[Param], syntax: &SyntaxNo
...
                     if param.kind() == ParamKind::MutRef && node.mut_token().is_some() =>
                 {
-                    ted::replace(node.syntax(), node.expr().unwrap().syntax());
+                    ted::replace(node.syntax(), node.expr()?.syntax());
                 }
                 Some(ast::Expr::RefExpr(node))
                     if param.kind() == ParamKind::SharedRef && node.mut_token().is_none() =>
                 {
-                    ted::replace(node.syntax(), node.expr().unwrap().syntax());
+                    ted::replace(node.syntax(), node.expr()?.syntax());
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/extract_function.rs:1949]: Let's turn these into `if let`s and do nothing for the `None` case.

File: crates/ide-assists/src/handlers/extract_function.rs [hunk 2]
```diff
@@ -1721,14 +1720,14 @@ fn make_body(
...
+) -> Option<ast::BlockExpr> {
     let ret_ty = fun.return_type(ctx);
     let handler = FlowHandler::from_ret_ty(fun, &ret_ty);
 
     let block = match &fun.body {
         FunctionBody::Expr(expr) => {
-            let expr = rewrite_body_segment(ctx, &fun.params, &handler, expr.syntax());
-            let expr = ast::Expr::cast(expr).unwrap();
+            let expr = rewrite_body_segment(ctx, &fun.params, &handler, expr.syntax())?;
+            let expr = ast::Expr::cast(expr)?;
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/extract_function.rs:1730]: Can't happen, so let's make this an expect

**Review by Veykril — CHANGES_REQUESTED** [review]
Also a pretty expensive assist, so we should make sure this stays lazy

File: crates/ide-assists/src/handlers/extract_function.rs [hunk 3]
```diff
@@ -103,50 +103,49 @@ pub(crate) fn extract_function(acc: &mut Assists, ctx: &AssistContext<'_>) -> Op
...
+    let old_indent = fun.body.indent_level();
+
+    let fn_def = match fun.self_param_adt(ctx) {
+        Some(adt) if anchor == Anchor::Method && !has_impl_wrapper => {
+            let fn_def = format_function(ctx, module, &fun, old_indent, new_indent + 1)?;
+            generate_impl_text(&adt, &fn_def).replace("{\n\n", "{")
+        }
+        _ => format_function(ctx, module, &fun, old_indent, new_indent)?,
+    };
+
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/extract_function.rs:141]: Given the other reviews, all of this can stay in the builder closure

**Review by Veykril — CHANGES_REQUESTED** [review]

**Comment by alibektas** [comment]
It seems like I did more damage than fixing stuff with these last commits :smile:. So now I basically took the original version ( without actually reverting it) and changed `unwrap`s to `expect`s

**Review by Veykril — APPROVED** [review]

**Pull request merged**